util/request_source.go

```diff
@@ -27,11 +27,23 @@ const (
 	InternalTxnMeta = InternalTxnOthers
 )
 
+// explicit source types.
+const (
+	ExplicitTypeDefault    = ""
+	ExplicitTypeLightning  = "lightning"
+	ExplicitTypeBR         = "br"
+	ExplicitTypeDumpling   = "dumpling"
+	ExplicitTypeBackground = "background"
+)
```

Comment:
I think they are both a type it's more convenient
```suggestion
type ExplicitType string

const (
	ExplicitTypeDefault    ExplicitType = ""
	ExplicitTypeLightning  ExplicitType = "lightning"
	ExplicitTypeBR         ExplicitType = "br"
	ExplicitTypeDumpling   ExplicitType = "dumpling"
	ExplicitTypeBackground ExplicitType = "background"
)
```

And then we can point which is background type like:
```
func IsBackgroundRequest(sourceType ExplicitType) bool {
	return sourceType != ExplicitTypeDefault
}
```
What do you think?

Reply:
It should judge by the resource group setting I think.